crates/ra_assists/src/handlers/change_visibility.rs

```diff
@@ -72,6 +77,125 @@ fn add_vis(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
     })
 }
 
+fn add_vis_to_referenced_module_def(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let path: ast::Path = ctx.find_node_at_offset()?;
+    let path_res = ctx.sema.resolve_path(&path)?;
+    let def = match path_res {
+        PathResolution::Def(def) => def,
+        _ => return None,
+    };
+
+    let current_module = ctx.sema.scope(&path.syntax()).module()?;
+    let target_module = def.module(ctx.db)?;
+
+    let vis = target_module.visibility_of(ctx.db, &def)?;
+    if vis.is_visible_from(ctx.db, current_module.into()) {
+        return None;
+    };
+    let target_name;
+
+    let (offset, target, target_file) = match def {
+        hir::ModuleDef::Function(f) => {
+            target_name = Some(f.name(ctx.db));
+            offset_target_and_file_id(ctx.db, f)
+        }
+        hir::ModuleDef::Adt(adt) => {
+            target_name = Some(adt.name(ctx.db));
+            match adt {
+                hir::Adt::Struct(s) => offset_target_and_file_id(ctx.db, s),
+                hir::Adt::Union(u) => offset_target_and_file_id(ctx.db, u),
+                hir::Adt::Enum(e) => offset_target_and_file_id(ctx.db, e),
+            }
+        }
+        hir::ModuleDef::Const(c) => {
+            target_name = c.name(ctx.db);
+            offset_target_and_file_id(ctx.db, c)
+        }
+        hir::ModuleDef::Static(s) => {
+            target_name = s.name(ctx.db);
+            offset_target_and_file_id(ctx.db, s)
+        }
+        hir::ModuleDef::Trait(t) => {
+            target_name = Some(t.name(ctx.db));
+            offset_target_and_file_id(ctx.db, t)
+        }
+        hir::ModuleDef::TypeAlias(t) => {
+            target_name = Some(t.name(ctx.db));
+            offset_target_and_file_id(ctx.db, t)
+        }
+        hir::ModuleDef::Module(m) => {
+            target_name = m.name(ctx.db);
+            let in_file_source = m.declaration_source(ctx.db)?;
+            let file_id = in_file_source.file_id.original_file(ctx.db);
+            let syntax = in_file_source.value.syntax();
+            (vis_offset(syntax), syntax.text_range(), file_id)
+        }
+        // Enum variants can't be private, we can't modify builtin types
+        hir::ModuleDef::EnumVariant(_) | hir::ModuleDef::BuiltinType(_) => return None,
+    };
+
+    // FIXME if target is in another crate, add `pub` instead of `pub(crate)`
```

Comment:
hmm this shouldn't be too hard, right? you just need to compare the crate of `target_module` with the one of `current_module`. we could also just not offer the assist in that case for now.

Reply:
true, that was very easy :)